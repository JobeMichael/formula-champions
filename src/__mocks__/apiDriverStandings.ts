export const apiDriverStandings = {
  MRData: {
    StandingsTable: {
      season: "2006",
      StandingsLists: [
        {
          season: "2006",
          DriverStandings: [
            {
              position: "1",
              points: "134",
              wins: "7",
              Driver: {
                driverId: "alonso",
                code: "ALO",
                givenName: "Fernando",
                familyName: "Alonso",
                nationality: "Spanish",
              },
              Constructors: [
                {
                  name: "Renault",
                },
              ],
            },
          ],
        },
      ],
    },
  },
};
