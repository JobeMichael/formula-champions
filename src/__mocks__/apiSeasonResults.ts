export const apiSeasonResults = {
  MRData: {
    xmlns: "http://ergast.com/mrd/1.4",
    series: "f1",
    url: "http://ergast.com/api/f1/2005/results/1.json",
    limit: "30",
    offset: "0",
    total: "19",
    RaceTable: {
      season: "2005",
      position: "1",
      Races: [
        {
          season: "2005",
          round: "1",
          url: "http://en.wikipedia.org/wiki/2005_Australian_Grand_Prix",
          raceName: "Australian Grand Prix",
          Circuit: {
            circuitId: "albert_park",
            url: "http://en.wikipedia.org/wiki/Melbourne_Grand_Prix_Circuit",
            circuitName: "Albert Park Grand Prix Circuit",
            Location: {
              lat: "-37.8497",
              long: "144.968",
              locality: "Melbourne",
              country: "Australia",
            },
          },
          date: "2005-03-06",
          time: "14:00:00Z",
          Results: [
            {
              number: "6",
              position: "1",
              positionText: "1",
              points: "10",
              Driver: {
                driverId: "fisichella",
                code: "FIS",
                url: "http://en.wikipedia.org/wiki/Giancarlo_Fisichella",
                givenName: "Giancarlo",
                familyName: "Fisichella",
                dateOfBirth: "1973-01-14",
                nationality: "Italian",
              },
              Constructor: {
                constructorId: "renault",
                url: "http://en.wikipedia.org/wiki/Renault_in_Formula_One",
                name: "Renault",
                nationality: "French",
              },
              grid: "1",
              laps: "57",
              status: "Finished",
              Time: {
                millis: "5057336",
                time: "1:24:17.336",
              },
              FastestLap: {
                rank: "2",
                lap: "55",
                Time: {
                  time: "1:25.994",
                },
              },
            },
          ],
        },
      ],
    },
  },
};
