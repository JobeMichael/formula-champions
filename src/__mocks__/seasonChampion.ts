export const seasonChampion = {
  season: "2006",
  points: "134",
  wins: "7",
  name: "Fernando Alonso",
  nationality: "Spanish",
  team: "Renault",
  driverId: "alonso",
};
