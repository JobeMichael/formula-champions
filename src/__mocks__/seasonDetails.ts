export const seasonDetails = {
  date: "2005-03-06",
  name: "Giancarlo Fisichella",
  nationality: "Italian",
  raceName: "Australian Grand Prix",
  season: "2005",
  team: "Renault",
  time: "1:24:17.336",
  driverId: "fisichella",
};
