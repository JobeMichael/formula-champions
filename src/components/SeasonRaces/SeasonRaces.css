.star {
  color: #c3c322;
  font-size: 26px;
  position: absolute;
  top: 0px;
}

.table td,
.table th {
  padding: 0.65rem !important;
  vertical-align: middle;
  position: relative;
}
