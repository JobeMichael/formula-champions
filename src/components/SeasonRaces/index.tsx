import React from "react";
import { Table } from "react-bootstrap";
import "./SeasonRaces.css";

const SeasonRaces = ({ data, driverId: ChampionId }: any) => (
  <Table responsive striped bordered hover>
    <thead>
      <tr>
        <th>Race</th>
        <th>Driver</th>
        <th>Team</th>
        <th>Nationality</th>
        <th>Time</th>
      </tr>
    </thead>
    <tbody>
      {data.map(
        ({
          name,
          nationality,
          team,
          season,
          time,
          raceName,
          driverId,
        }: any) => (
          <tr key={raceName}>
            <td>{raceName}</td>
            <td>
              {name}{" "}
              {driverId === ChampionId && (
                <span className="star">&#10026;</span>
              )}
            </td>
            <td>{team}</td>
            <td>{nationality}</td>
            <td>{time}</td>
          </tr>
        )
      )}
    </tbody>
  </Table>
);

export default SeasonRaces;
