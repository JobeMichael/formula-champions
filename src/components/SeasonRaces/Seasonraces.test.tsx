import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import SeasonRaces from ".";
import { seasonDetails } from "../../__mocks__/seasonDetails";

describe("<SeasonRaces/>", () => {
  it("renders SeasonRaces correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SeasonRaces data={[seasonDetails]} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("highlight the row when driver is the champion", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SeasonRaces data={[seasonDetails]} driverId="fisichella" />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
