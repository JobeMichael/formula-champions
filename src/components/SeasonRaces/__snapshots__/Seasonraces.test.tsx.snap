// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeasonRaces/> highlight the row when driver is the champion 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Race
          </th>
          <th>
            Driver
          </th>
          <th>
            Team
          </th>
          <th>
            Nationality
          </th>
          <th>
            Time
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Australian Grand Prix
          </td>
          <td>
            Giancarlo Fisichella
            <span
              class="star"
            >
              âœª
            </span>
          </td>
          <td>
            Renault
          </td>
          <td>
            Italian
          </td>
          <td>
            1:24:17.336
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`<SeasonRaces/> renders SeasonRaces correctly 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Race
          </th>
          <th>
            Driver
          </th>
          <th>
            Team
          </th>
          <th>
            Nationality
          </th>
          <th>
            Time
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Australian Grand Prix
          </td>
          <td>
            Giancarlo Fisichella
          </td>
          <td>
            Renault
          </td>
          <td>
            Italian
          </td>
          <td>
            1:24:17.336
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
